src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelDiscoveryEventsHandler.cs

```diff
@@ -86,11 +96,24 @@ public void HandleDiscoveryComplete(DiscoveryCompleteEventArgs discoveryComplete
                     LastDiscoveredTests = null
                 };
 
+                // Collecting Final Discovery State
+                this.requestData.MetricsCollection.Add(TelemetryDataConstants.DiscoveryState, isAborted ? "Aborted" : "Completed");
+
+                // Collect Aggregated Metrics Data
+                var aggregatedDiscoveryDataMetrics = discoveryDataAggregator.GetAggregatedDiscoveryDataMetrics();
+                if (aggregatedDiscoveryDataMetrics != null && aggregatedDiscoveryDataMetrics.Count != 0)
+                {
+                    foreach (var aggregatedRunDataMetric in aggregatedDiscoveryDataMetrics)
```

Comment:
DRY:  Can we have method in the MetricsCollection to add values from other dictionary?

Reply:
Yes, will refactor in publish logic in next PR.